src/os/README.md

```diff
@@ -50,19 +50,23 @@ inspiration for several decisions in this library.
 - do not use `OS\ErrnoException` if the error condition would not be indicated
   by the `errno` variable in C. Consider adding another similar class, e.g.
   add `OS\HErrnoException` if you want to report an error exposed via `h_errno`
-- add and use Hack classes (not type aliases) for 'handle'-like parameters and
-  return values, e.g. `OS\open()` returns a `HH\Lib\FileDescriptor` instead of
-  an `int`; as well as aiding type safety, this prevents requests from interfering
-  with resources that belong to another request.
+- use `keyset<Flag>` to represent a C bit set, where `Flag` is a Hack `enum`.
+- add and use Hack classes (not type aliases) for long-lived 'handle'-like
+  parameters and return values, e.g. `OS\open()` returns a
+  `HH\Lib\FileDescriptor` instead of an `int`; as well as aiding type safety,
+  this prevents requests from interfering with resources that belong to another
+  request.
 - Avoid `inout` parameters; return tuples instead. For example, prefer
   `function mkstemp(string $pattern): (FileDescriptor, string)` to
   `function mkstemp(inout string $in_pattern_out_path): FileDescriptor`
   - this can aid for common use by allowing string literals, rather than
     requiring otherwise-unused locals
-  - if the primary purpose of a function is to mutate its parameters, this
-    should be considered case-by-case; for example, if we were to add something
-    similar to `array_pop()`, it probably should take an `inout` container - but
-    probably shouldn't be in `OS\`
+  - if the primary purpose of a set of functions is to create, mutate and
+    destroy a short-lived opaque C pointer, they should be exposed as a `shape`
+    or `vec`. See [Appendix: opaque C pointer
+    encoding](#appendix-opaque-c-pointer-encoding) section for more detail.
+- Functions that are not available in all the Hack supported operating systems
+  should put into separate namespaces, e.g. `HH\Lib\OS\Bsd` or `HH\Lib\OS\Linux`.
```

Comment:
This is an unfinished discussion; it also implies that if we support more operating systems in the future, that we will move things out of the main `OS\` namespace, which I don't think is the case (or desirable, from a BC perspective)

Reply:
I split this item into another PR. #171